required_scripts/sigprofiler_7595614.py

```diff
@@ -124,10 +124,11 @@ def main():
                 os.remove(os.path.join(args.vcfpath+"/output/ID", f))
             os.rmdir(args.vcfpath+"/output/ID")
 
-    cpus_per_task = max(int(args.threads/num_tasks),1)
-    with ThreadPoolExecutor(max_workers=3) as e:
-        for sigClass, sigContext in sig_list:
-            e.submit(extractSignatures, args.output, args.vcfpath, args.genome, args.project, sigClass, sigContext, args.exome, cpus_per_task)
+    if num_tasks > 0:
```

Comment:
Isn't `numtask`  defined earlier in the code? Why checking if >0;

Reply:
It's defined at 0, then based on the number of signature types found it will increment the num_tasks. Hence if no tasks are generated then we will still be 0.